index.js

```diff
@@ -91,6 +100,28 @@ function _insertCredentials(username, userhash) {
 let cachedAgents;
 
 /**
+ * Generates and maps an internal username to the given username
+ *
+ * To protect against username collisions, this function takes
```

Comment:
Can we mention parallelisation somewhere?
E.g.: `To protect against username collisions during test parallelisation, this function ....`